### PR TITLE
fix: improve agent loading and error handling

### PR DESCRIPTION
**Pull request by XiangZhang-zx**
- Add debug logs to track agent loading
- Fix config handling in agent manager
- Update seeact_demo_agent config format
- Improved agent loading mechanism to support SeeAct
- Added configuration validation for Agents

The code now:
- Shows better error messages
- Loads agents more reliably
- Keeps full config when packaging

### CODE REVIEW

File: hooks/modules/agent.py
[the remark this answers is not in the document]
Comment (dongyuanjushi) [reply hooks/modules/agent.py:1]: Is this hook outside the aios folder necessary or redundant?

[the remark this answers is not in the document]
Comment (XiangZhang-zx) [reply hooks/modules/agent.py:1]: The run_agent function in hooks/modules/agent.py is not used in the current codebase. It appears to be redundant. I added some debug logs but found it wasn't being called anywhere. We can consider removing it.

[the remark this answers is not in the document]
Comment (dongyuanjushi) [reply hooks/modules/agent.py:1]: okay, lets remove this folder and merge

[the remark this answers is not in the document]
Comment (XiangZhang-zx) [reply hooks/modules/agent.py:1]: The run_agent function in hooks/modules/agent.py is used when we run the kernel in kernel.py. It is called by useFactory in aios.hooks.modules.agent. I tried removing it, but the system didn't work, so it is necessary.  image

**Review by dongyuanjushi — APPROVED** [review]
The Gemini API seems out of quota for this. We probably need to provide ollama models for testing this.

**Pull request merged**